src/object/cli_ec.c

```diff
@@ -1254,25 +1293,20 @@ obj_ec_recx_reasb(daos_iod_t *iod, d_sg_list_t *sgl,
 		last = tgt_recx_idxs[i] + tgt_recx_nrs[i];
 	}
 	oiod->oiod_nr = idx;
-	tmp_nr = update ? obj_ec_data_tgt_nr(oca) : tgt_nr;
 	for (i = 0, rec_nr = 0, last = 0; i < tgt_nr; i++) {
 		if (tgt_recx_nrs[i] == 0)
 			continue;
-		if ((*single_data_tgt) && (i < tmp_nr)) {
-			if (*data_tgt_idx == -1)
-				*data_tgt_idx = i;
-			else if (*data_tgt_idx != i)
-				*single_data_tgt = false;
-		}
 		siod = &oiod->oiod_siods[tidx[i]];
-		siod->siod_tgt_idx = i;
+		siod->siod_tgt_idx = obj_ec_shard_idx(dkey_hash, oca, i);
 		siod->siod_idx = tgt_recx_idxs[i];
 		siod->siod_nr = tgt_recx_nrs[i];
+		D_DEBUG(DB_TRACE, "i %d tgt %u idx %u nr %u, start "DF_U64
+			" tgt_recx %u/%u\n", i, siod->siod_tgt_idx, siod->siod_idx,
+			siod->siod_nr, obj_ec_shard_idx(dkey_hash, oca, 0),
+			tgt_recx_idxs[i], tgt_recx_nrs[i]);
```

Comment:
Sure, but EC_TRACE would not be enabled without re-compile, so it would be an issue if we want customer to collect the log. 

So I would rather unify the log trace format, i.e. we can use DD_MASK to control that. thanks.